### DISCUSSION TITLE
NoSuchMethod exception when using shaded Configurate in BungeeCord

### DISCUSSION THREAD

**Discussion by Andre601** [SpongePowered/Configurate#191] [Support] [answered]
When I updated to Configurate 4.0.0 do I now receive a warning with Exception from BungeeCord with a NoSuchMethodException.

Configurate did work fine before and the only changes I made are updating the version and the `<pattern>` to shade.

Here's the error: https://paste.helpch.at/raquzimahi.coffeescript

Lines where the error points to (In order of appearing:
- https://github.com/Andre601/OneVersionRemake/blob/master/core/src/main/java/com/andre601/oneversionremake/core/files/ConfigHandler.java#L70
- https://github.com/Andre601/OneVersionRemake/blob/master/core/src/main/java/com/andre601/oneversionremake/core/OneVersionRemake.java#L68
- https://github.com/Andre601/OneVersionRemake/blob/master/core/src/main/java/com/andre601/oneversionremake/core/OneVersionRemake.java#L42

The file is loaded and handled by a ConfigHandler class:
https://github.com/Andre601/OneVersionRemake/blob/master/core/src/main/java/com/andre601/oneversionremake/core/files/ConfigHandler.java

Any help is appreciated.

**Comment by zml2008** [comment, marked as answer]
As of v4.0.0, Configurate does some deeper integration with SnakeYaml, which results in us being stricter on which SnakeYaml versions we support. It looks like Bungee ships an outdated version of SnakeYaml.

You'll want to either get BungeeCord to update the version of SnakeYaml they ship, or shade the library yourself in addition to Configurate.

**Reply by Andre601** [reply]
Current version for BungeeCord is 1.27 and from looking at https://github.com/SpongePowered/Configurate/blob/master/format/yaml/build.gradle.kts does it seem like that it *should* be fine since that is `1.+` (`1.x`)

**Comment by Andre601** [comment]
Pumping BungeeCord dependency to `1.16-R0.5-SNAPSHOT` and also making sure to use latest build fixed the problem.

**Reply by zml2008** [reply]
glad to hear that fixed things :)
